src/Uno.Foundation/Uno.Core.Extensions/Uno.Core.Extensions/StringExtensions.cs

```diff
@@ -87,22 +88,23 @@ public static LineSplitEnumerator SplitLines(this ReadOnlySpan<char> instance)
 			return new LineSplitEnumerator(instance);
 		}
 
-		public static bool IsNullOrEmpty(this string instance)
+		
+		public static bool IsNullOrEmpty([NotNullWhen(false)] this string instance)
 		{
 			return string.IsNullOrEmpty(instance);
 		}
-
-		public static bool IsNullOrWhiteSpace(this string instance)
+		
+		public static bool IsNullOrWhiteSpace([NotNullWhen(false)] this string instance)
 		{
 			return string.IsNullOrWhiteSpace(instance);
 		}
 
-		public static bool HasValue(this string instance)
+		public static bool HasValue([NotNullWhen(true)] this string instance)
 		{
 			return !string.IsNullOrWhiteSpace(instance);
 		}
 
-		public static bool HasValueTrimmed(this string instance)
+		public static bool HasValueTrimmed([NotNullWhen(true)] this string instance)
 		{
 			return !string.IsNullOrWhiteSpace(instance);
 		}
```

Comment:
These are the same, and that's very confusing.

Reply:
I agree :) It's legacy and probably should be cleaned.